hibernate-core/src/main/java/org/hibernate/boot/beanvalidation/BeanValidationEventListener.java

```diff
@@ -121,6 +124,17 @@ public boolean onPreUpsert(PreUpsertEvent event) {
 		return false;
 	}
 
+	@Override
+	public void onPreUpdateCollection(PreCollectionUpdateEvent event) {
+		final Object entity = castNonNull( event.getCollection().getOwner() );
+		validate(
+				entity,
+				event.getSession().getEntityPersister( event.getAffectedOwnerEntityName(), entity ),
+				event.getFactory(),
+				GroupsPerOperation.Operation.UPDATE
+		);
+	}
```

Comment:
hmm interesting 😃 , what if we have a deeper entity tree, i.e. 
```java
@Entity
class Parent {
   @NotBlank
   String someString;
   @OneToOne
   TestEntity entity
}
```
and then we make the change through this `Parent` updating both `someString` and collection in `TestEntity` will we get 2 exceptions or not ...

Reply:
Just tested this, and it would not: the validation fails but only `someString` is reported as invalid. 

Though note that even if updating the set property by changing its value (i.e. `parent.entity.notEmptySet = null`) instead of using the setter, only 1 constraint validation error is reported as well, this time with the set property itself (I'm guessing this depends on the order of entity update actions in the execution queue).

Reply:
oh and another case when things might start behaving differently would be if you have other constraints in `TestEntity` on "regular" fields that are violated.. e.g.:
```java
class TestEntity {
  @Id
  private Long id;

  @ManyToMany(cascade = CascadeType.PERSIST)
  @NotEmpty
  private Set<ChildEntity> notEmptySet;

  @ElementCollection
  @Size(min = 1)
  private List<String> minSizeList;

  @Future 
  private LocalDate date = LocalDate.now();
}
```

and if you try setting date to the past and updating the collection, we'd validate the test entity twice, no?

Reply:
> and if you try setting date to the past and updating the collection, we'd validate the test entity twice, no?

No, actually in that case the original `PreUpdateEvent` would be triggered for `TestEntity` and we would catch both constraint violations (producing 2 errors). Let me add a test for that actually, thanks for pointing it out.

Reply:
that's nice then 🎉 😃 
thanks!